optimum/neuron/utils/hub_neuronx_cache.py

```diff
@@ -0,0 +1,229 @@
+
+    def _rel_path(self, path):
+        # Remove the default cache url from the path
+        if path.startswith(self.default_cache.cache_path):
+            return path[len(self.default_cache.cache_path) :].lstrip("/")
+
+    def exists(self, path):
+        # Always prioritize the default cache
+        if self.default_cache.exists(path):
+            return True
+        rel_path = self._rel_path(path)
+        exists = self.api.file_exists(self.repo_id, rel_path)
+        if not exists:
+            logger.warning(
+                f"{rel_path} not found in {self.repo_id}: the corresponding graph will be recompiled."
+                " This may take up to one hour for large models."
+            )
+        return exists
+
+    def download_file(self, filename, dst_path):
+        # Always prioritize the default cache for faster retrieval
+        if self.default_cache.exists(filename):
+            self.default_cache.download_file(filename, dst_path)
+        else:
+            rel_filename = self._rel_path(filename)
+            local_path = self.api.hf_hub_download(self.repo_id, rel_filename)
+            os.symlink(local_path, dst_path)
+            logger.info(f"Fetched cached {rel_filename} from {self.repo_id}")
+
+    def synchronize(self):
+        if isinstance(self.default_cache, CompileCacheS3):
+            raise ValueError("HuggingFace hub compiler cache synchronization is not supported for S3.")
+        logger.info(f"Synchronizing {self.repo_id} Hub cache with {self.default_cache.cache_path} local cache")
+        self.api.upload_folder(
+            repo_id=self.repo_id,
+            folder_path=self.default_cache.cache_path,
+            commit_message="Synchronizing local compiler cache.",
+            ignore_patterns="lock",
+        )
+        logger.info("Synchronization complete.")
+
+    def upload_file(self, cache_path, src_path):
+        # Only upload to the default cache: use synchronize to populate the Hub cache
+        self.default_cache.upload_file(cache_path, src_path)
+
+    def upload_string_to_file(self, cache_path, data):
+        # Only upload to the default cache: use synchronize to populate the Hub cache
+        self.default_cache.upload_string_to_file(cache_path, data)
+
+    def download_file_to_string(self, filename, limit=None):
+        # Always prioritize the default cache for faster retrieval
+        if self.default_cache.exists(filename):
+            return self.default_cache.download_file_to_string(filename, limit)
+        rel_filename = self._rel_path(filename)
+        local_path = self.api.hf_hub_download(self.repo_id, rel_filename)
+        with open(local_path, "rb") as f:
+            s = f.read().decode(errors="replace")
+        logger.info(f"Fetched cached {rel_filename} from {self.repo_id}")
+        return s
+
+
+def get_hub_cache():
+    HUB_CACHE = "aws-neuron/optimum-neuron-cache"
+    return os.getenv("CUSTOM_CACHE_REPO", HUB_CACHE)
```

Comment:
I think you can use the `optimum.neuron.utils.cache_utils.get_hf_hub_cache_repos` function for this purpose. 
It will also check the locally set up HF repo.

Reply:
I can't because it does not return anything on staging.

Reply:
But that's expected no? On staging the `aws-neuron/optimum-neuron-cache` repo does not exist.

Reply:
[It can also take into account custom cache repos](https://github.com/huggingface/optimum-neuron/blob/main/optimum/neuron/utils/cache_utils.py#L193-L195). But anyways that's not a big deal, just wanted to let you know that it exists.